src/mavedb/routers/experiments.py

```diff
@@ -133,7 +145,9 @@ def fetch_experiment(
         raise HTTPException(status_code=404, detail=f"Experiment with URN {urn} not found")
 
     assert_permission(user_data, item, Action.READ)
-    return item
+    updated_experiment = enrich_experiment_with_num_score_sets(item, user_data)
+
+    return updated_experiment
```

Comment:
We can return the object directly
```suggestion
    return enrich_experiment_with_num_score_sets(item, user_data)
```